packages/knip/src/typescript/resolveModuleNames.ts

```diff
@@ -91,12 +91,29 @@ export function createCustomModuleResolver(
 
       return {
         resolvedFileName,
-        extension: virtualFileExtensions.includes(ext) ? ts.Extension.Js : ext,
+        extension: getExtension(resolvedFileName),
```

Comment:
Why are we using a function with `customCompilerExtensions.some` and `customCompilerExtensions.includes`? Maybe we can make it consistent/fast with a Set + `Set.has`?

Reply:
Good point. I removed the function and made it more consistent.